lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -106,12 +106,35 @@ e2function void setName( string name )
 	e:SetOverlayText(name)
 end
 
--- Get the name of another E2
+-- Set the name of a entity (component name if not E2). Thanks CaptainPRICE for idea and basic implementation
+e2function void entity:setName( string name )
+	if not IsValid(this) or E2Lib.getOwner(self, this) ~= self.player then return end
+	if this:GetClass() == "gmod_wire_expression2" then
+		if this.name == name then return end
```

Comment:
Is this early-out purely here for optimisation reasons? It makes the "generic" name logic below harder to reason about.